source/common/http/http2/codec_impl.cc

```diff
@@ -260,7 +298,13 @@ ConnectionImpl::StreamImpl* ConnectionImpl::getStream(int32_t stream_id) {
 }
 
 int ConnectionImpl::onData(int32_t stream_id, const uint8_t* data, size_t len) {
-  getStream(stream_id)->pending_recv_data_.add(data, len);
+  StreamImpl* stream = getStream(stream_id);
+  stream->pending_recv_data_.add(data, len);
```

Comment:
Does the H2 stream flow control window save us from infinite buffering in `pending_recv_data_` here? If so, maybe add a comment.